src/ansiblelint/prerun.py

```diff
@@ -158,23 +158,47 @@ def prepare_environment() -> None:
     _prepare_ansible_paths()
 
 
+def _get_galaxy_role_ns(galaxy_infos: Dict[str, Any]) -> str:
+    """Compute role namespace from meta/main.yml, including trailing dot."""
+    role_namespace = galaxy_infos.get('namespace', None)
+    if not role_namespace:
+        role_namespace = galaxy_infos.get('author', None)
+    # if there's a space in the name space, it's likely author name
+    # and not the galaxy login, so act as if there was no namespace
+    if re.match(r"^\w+ \w+", role_namespace):
+        role_namespace = ""
+    else:
+        role_namespace = f"{role_namespace}."
+    return role_namespace
+
+
+def _get_galaxy_role_name(galaxy_infos: Dict[str, Any]) -> str:
```

Comment:
I am surprised this passes mypy because the correct return type is `Optional[str]` and not str based on the fact I see the default of `None`.